docs/guides/ngrok-for-ingress-to-services-in-a-consul-service-mesh-on-kubernetes.md

```diff
@@ -0,0 +1,312 @@
+---
+title: ngrok for Ingress to Services in a Consul Service Mesh on Kubernetes
+---
+
+# Using ngrok Ingress Controller with Hashicorp's Consul Service Mesh
+
+## Introduction
+
+The [ngrok Ingress Controller for Kubernetes](https://github.com/ngrok/kubernetes-ingress-controller) is our official open-source controller for adding public and secure ingress traffic to your k8s services. It works any Cloud, local, or On-Prem Kubernetes cluster to provide ingress to your services no matter what the network configurations are, as long as it has outbound access to the ngrok service. This allows ngrok to be portable and work seamlessly across any different type of infrastructure.
+
+Consul is a secure and resilient service mesh that provides service discovery, configuration, and segmentation functionality. Consul Connect provides service-to-service connection authorization and encryption using mutual TLS, automatically enabling TLS for all Connect services. Consul can be used with Kubernetes to provide a service mesh for your Kubernetes cluster.
+
+Together, Consul can provide a robust and secure way for Services within a cluster to communicate, while ngrok can seamlessly and securely provide public ingress to them. This guide will walk you through setting up a Consul Service Mesh on Kubernetes and then using the ngrok Ingress Controller to provide ingress to your services to illustrate how they can work together.
```

Comment:
> Together, Consul can provide a robust and secure way for Services within a cluster to communicate, while ngrok can seamlessly and securely provide public ingress to them.

Together, Consul _provides_ a robust and secure way for Services within a cluster to communicate, while ngrok seamlessly and securely _provides_ public ingress to them.